### PR TITLE
Pagination support for lists

### PR DESCRIPTION
**Pull request by jacobgarder**
The ZScaler API does only return maximum of 100 entries per request by default. To be able to list more entries, I would like support for pagination in this SDK. 

This PR contains fixes for the listings in location and traffic.  Please let me know what you think!

### CODE REVIEW

**Comment by mitchos** [comment]
Thanks for the pull request @jacobgarder!

pyZscaler should absolutely support the pagination so agree with you that it needs to be added. 

If we just pass the pagination parameters directly to ZIA, the developer would need to make multiple API calls within their own code or create a loop to handle fetching each page. I think pyZscaler should do the heavy lifting here.

I'm not against adding your changes in the short-term, but we'll need a more robust solution going forward.

I'd like to see an iterator that uses the default pagination values but can takes some additional params from the user if required.

e.g.

`zia.locations.list(page_size=100, pages=3)
`

The Restfly library that we're using for pyZscaler has a [pagination iterator](https://restfly.readthedocs.io/en/latest/user/pagination.html) that we could use. 

If you want to work on this let me know and I'll leave this PR open until you finish that, or if you don't have the time I'll go through review of your changes and merge in the short-term - then develop the iterator for a future release.

Thoughts?

**Comment by jacobgarder** [comment]
It's my pleasure @mitchos! I think I could use a good Zscaler-SDK and thought of making one for myself when I found your project. :)

I agree, a iterator is much neater! Restfly seems pretty nifty (have not used it before). One question that arises is that if the list-function then should return the iterator rather than a list.. That would change the behavior of the function though, so I look into implementing it, still returning a full list by default.

**Comment by jacobgarder** [comment]
Updated the PR now leveraging an Iterator class in locations and traffic instead. Still returning a list of all elements. I think that is probably what most people want.
Added a limit in the pytest as well as the mock response will never come to an end.

**Comment by mitchos** [comment]
Okay that's exactly what I had in my head for that. Great work here and thank you!

I've created a new feature branch based off develop called `feature/zia-pagination`.

Let me know if you're okay with me moving your pull request there, that way we can add this support to all the get endpoints that support pagination, update tests etc.

File: pyproject.toml [hunk 1]
```diff
@@ -4,7 +4,7 @@ version = "0.8.0"
 description = "A python SDK for the Zscaler API."
 authors = ["Mitch Kelly <me@mkelly.dev>"]
 license = "MIT"
-readme = "README.md"
```
Comment (mitchos) [inline pyproject.toml:7]: Just noticed that I hadn't added this file to git. README.md was added to have a condensed version of GitHub's README.rst for PyPi builds. I'll add this to `develop` today. Keep this as README.md.

File: pyzscaler/zia/locations.py [hunk 2]
```diff
@@ -1,21 +1,29 @@
...
+
+from pyzscaler.utils import Iterator, snake_to_camel
 
 
 class LocationsAPI(APIEndpoint):
-    def list_locations(self):
+    def list_locations(self, **kwargs):
         """
-        Returns a list of configured locations.
+        Return list of locations.
```
Comment (mitchos) [inline pyzscaler/zia/locations.py:10]: Quality/Grammar: Returns a list of locations.

File: pyzscaler/zia/locations.py [hunk 3]
```diff
@@ -1,21 +1,29 @@
...
 
+        Keyword Args:
+            **max_items (int, optional):
+                The maximum number of items to return before stopping iteration.
+            **max_pages (int, optional):
+                The maximum number of pages to request before throwing stopping iteration.
```
Comment (mitchos) [inline pyzscaler/zia/locations.py:16]: Quality/Grammar: The maximum number of pages to request before stopping iteration.

File: pyzscaler/zia/locations.py [hunk 4]
```diff
@@ -91,10 +99,18 @@ def list_sub_locations(self, location_id: str):
...
-    def list_locations_lite(self):
+    def list_locations_lite(self, **kwargs):
         """
         Returns only the name and ID of all configured locations.
 
+        Keyword Args:
+            **max_items (int, optional):
+                The maximum number of items to return before stopping iteration.
+            **max_pages (int, optional):
+                The maximum number of pages to request before throwing stopping iteration.
```
Comment (mitchos) [inline pyzscaler/zia/locations.py:110]: Quality/Grammar: The maximum number of pages to request before stopping iteration.

File: pyzscaler/zia/traffic.py [hunk 5]
```diff
@@ -1,21 +1,30 @@
...
-    def list_gre_tunnels(self):
+    def list_gre_tunnels(self, **kwargs):
         """
         Returns the list of all configured GRE tunnels.
 
+        Keyword Args:
+            **max_items (int, optional):
+                The maximum number of items to return before stopping iteration.
+            **max_pages (int, optional):
+                The maximum number of pages to request before throwing stopping iteration.
```
Comment (mitchos) [inline pyzscaler/zia/traffic.py:16]: Quality/Grammar: The maximum number of pages to request before stopping iteration.

File: pyzscaler/zia/traffic.py [hunk 6]
```diff
@@ -387,10 +396,18 @@ def get_closest_diverse_vip_ids(self, ip_address: str):
...
-    def list_vpn_credentials(self):
+    def list_vpn_credentials(self, **kwargs):
         """
         Returns the list of all configured VPN credentials.
 
+        Keyword Args:
+            **max_items (int, optional):
+                The maximum number of items to return before stopping iteration.
+            **max_pages (int, optional):
+                The maximum number of pages to request before throwing stopping iteration.
```
Comment (mitchos) [inline pyzscaler/zia/traffic.py:407]: Quality/Grammar: The maximum number of pages to request before stopping iteration.

**Review by mitchos — CHANGES_REQUESTED** [review]
Just some minor feedback to address and then we'll merge it into the feature branch.

File: pyzscaler/zia/locations.py (hunk 2 above)
Comment (jacobgarder) [reply pyzscaler/zia/locations.py:10]: I agree that we should use the more "imperative-style" of docstrings. But that did not seem to be the case in the rest of the code..

File: pyzscaler/zia/traffic.py (hunk 6 above)
Comment (jacobgarder) [reply pyzscaler/zia/traffic.py:407]: Thanks! Changeing the other line to "request" as well.

**Comment by jacobgarder** [comment]
> Just some minor feedback to address and then we'll merge it into the feature branch.

Thanks for the feedback, Mitch! Please go ahead and merge this into the feature branch. I will look into implement the Iterator for the other endpoints as well.

**Comment by jacobgarder** [comment]
Actually, there are quite many of the Zscaler API-endpoints that does not support pagination. I added support to those missing and pushed it to this branch. 
Feel free to merge it with main or if you still think we should keep it in development, that's fine as well.

**Review by mitchos — APPROVED** [review]
Looks good, approved by me.

**Pull request merged**